Chop nameprefix if too long

bart613 opened the pull request:
Nameprefix cannot be longer than 32 characters and given situation with long(ish) component name it's easy to cross 32 characters given we also have suffix `service-role` (this came out of testing via Hypothesis in another Terraform module which relies on this module).

Given this is name prefix for a specific resource (either role or role policy) I removed the suffix and added condition in which, if passed service name will cross 32 characters it'll get chopped at 32
characters.

bart613 committed:
Chop nameprefix if too long

Nameprefix cannot be longer than 32 characters and given situation with
long(ish) component name it's easy to cross 32 characters given we also
have suffix `service-role` (this came out of testing via Hypotesis in
another Terraform module which relies on this module).

Given this is name prefix for a specific resource (either role or role
policy) I removed the suffix and added condition in which, if passed
service name will cross 32 characters it'll get chopped at 32
characters.

keymon commented:
I wonder if it is possible to add a specific test for this case, maybe  by parametrising the name in the test module in `test/infra/main.tf` and adding a test case with a long prefix.

grahamlyons reviewed (approved):
Yeah, looks good. Same comment as @keymon; can it be tested?

bart613 commented:
Yes it can be tested and as soon as I tested it, it came back with boatloads of other issues (eg. target-group name cannot be longer than 32 characters).  Will implement full test and fix

bart613 committed:
Expand long_name test & chop target_group name

Target group name can be max 32 characters long.  Chop it at 32
characters.

Also, added test to create plan of a service that has got a very long
name (to verify specific resources will get chopped parameters)
